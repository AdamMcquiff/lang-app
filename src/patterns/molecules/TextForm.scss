.m-text-form {
  padding: { bottom: 1em; }
}