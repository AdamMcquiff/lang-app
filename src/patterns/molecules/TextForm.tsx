import * as React from 'react';
import Button from '../atoms/Button';

interface ITextFormProps {
    inputLabel: string,
    buttonTextLabel: string,
    buttonA11yLabel: string,
    onButtonClick(): void,
    onTextChange(e: any): void
}

class TextForm extends React.Component<ITextFormProps> {
    public render() {
        return (
            <form className="text-form">
                <label>
                    {this.props.inputLabel}
                    <input type="text" onChange={this.props.onTextChange} />
                </label>
                <Button a11yLabel={this.props.buttonA11yLabel} 
                        textLabel={this.props.buttonTextLabel}
                        onClick={this.props.onButtonClick}/>
            </form>
        )
    }

}

export default TextForm;