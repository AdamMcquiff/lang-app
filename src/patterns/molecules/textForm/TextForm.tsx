import * as React from 'react';
import { Button } from '../../atoms/button/Button';
import { TextInput } from '../../atoms/textInput/TextInput';
import './TextForm.scss';

interface ITextFormProps {
    inputLabel: string,
    buttonTextLabel: string,
    buttonA11yLabel: string,
    placeholder: string,
    hasBackground: boolean,
    hasButton: boolean,
    hasLabel: boolean,
    onSubmit(): void,
    onTextChange(e: any): void,
}

export class TextForm extends React.Component<ITextFormProps> {
    public static defaultProps = {
        buttonA11yLabel: 'Label me.',
        buttonTextLabel: 'Label me.',
        hasBackground: true,
        hasButton: true,
        hasLabel: true,
        onSubmit: () => null,
        onTextChange: () => null,
        placeholder: '',
    }
    
    public render() {
        return (
            <div className="m-text-form">
                <form onKeyDown={this.onSubmit}>
                    <TextInput textLabel={this.props.inputLabel}
                               a11yLabel={this.props.inputLabel} 
                               placeholder={this.props.placeholder}
                               hasLabel={this.props.hasLabel}
                               hasBackground={this.props.hasBackground}
                               onTextChange={this.props.onTextChange} 
                    />
                            
                    {this.props.hasButton &&
                        <Button a11yLabel={this.props.buttonA11yLabel} 
                                textLabel={this.props.buttonTextLabel}
                                onClick={this.props.onSubmit} 
                        />
                    }   
                </form>
            </div>
        )
    }

    private onSubmit = (event: any) => {
        if (event.key !== 'Enter') {
            return;
        }
    
        event.preventDefault();
        this.props.onSubmit();
        event.target.value = "";
    }
}