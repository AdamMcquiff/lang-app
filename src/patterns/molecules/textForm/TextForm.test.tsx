import { mount, shallow } from 'enzyme';
import * as React from 'react';
import TextForm from './TextForm';

describe('TextForm', () => {
    const props = {
        buttonA11yLabel: "Some label",
        buttonTextLabel: "Some label",
        hasButton: true,
        inputLabel: "Some label",
        onSubmit: jest.fn(),
        onTextChange: jest.fn(),
    };

    it('renders without crashing', () => {
        const component = shallow(<TextForm {...props} />);
        expect(component).toMatchSnapshot();
    });

    it('renders <button> element when hasButton is true', () => {
        const component = mount(<TextForm {...props} />);

        expect(component
            .find('.a-button')
            .exists()
        ).toBeTruthy();
    });

    it('does not render <button> element when hasButton is false', () => {
        const component = mount(<TextForm inputLabel={props.inputLabel} hasButton={false}/>);
        
        expect(component
            .find('.a-button')
            .exists()
        ).toBeFalsy();
    });

    it('onSubmit prop method is called after form is submitted via "enter" key press down', () => {
        const component = mount(<TextForm {...props} />);

        component
            .find('form')
            .simulate('keydown', { key: 'Enter' });

        expect(props.onSubmit).toBeCalled();
    });
});