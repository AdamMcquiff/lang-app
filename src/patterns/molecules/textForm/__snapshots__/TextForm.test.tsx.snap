// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextForm renders without crashing with all props 1`] = `
<div
  className="m-text-form"
>
  <form
    onKeyDown={[Function]}
  >
    <TextInput
      a11yLabel="Some label"
      hasBackground={true}
      hasLabel={true}
      onClick={[Function]}
      onTextChange={[MockFunction]}
      placeholder=""
      textLabel="Some label"
    />
    <Button
      a11yLabel="Some label"
      className=""
      onClick={[MockFunction]}
      textLabel="Some label"
    />
  </form>
</div>
`;

exports[`TextForm renders without crashing with only required props 1`] = `
<div
  className="m-text-form"
>
  <form
    onKeyDown={[Function]}
  >
    <TextInput
      a11yLabel="Some label"
      hasBackground={true}
      hasLabel={true}
      onClick={[Function]}
      onTextChange={[Function]}
      placeholder=""
      textLabel="Some label"
    />
    <Button
      a11yLabel="Label me."
      className=""
      onClick={[Function]}
      textLabel="Label me."
    />
  </form>
</div>
`;
